File: 【大三】网络与信息安全技术/实验/code/AES.js
// @ts-nocheck

function aesEncrypt(plaintext, key) {
  let result = ""
  for (let i = 0; i < plaintext.length; i++) {
    result += String.fromCharCode(
      plaintext.charCodeAt(i) ^ key.charCodeAt(i % key.length)
    )
  }
  return result
}

function aesDecrypt(cipher, key) {
  return aesEncrypt(cipher, key)
}

const text = "HelloAES"
const key = "key123"

const enc = aesEncrypt(text, key)
const dec = aesDecrypt(enc, key)

console.log("Plain:", text)
console.log("Encrypted:", enc)
console.log("Decrypted:", dec)

File: 【大三】网络与信息安全技术/实验/code/AES.cpp
#include <iostream>
#include <string>
using namespace std;

string aes_encrypt(string plaintext, string key) {
  string cipher = plaintext;
  for (int i = 0; i < plaintext.size(); i++) {
    cipher[i] = plaintext[i] ^ key[i % key.size()];
  }
  return cipher;
}

string aes_decrypt(string cipher, string key) {
  return aes_encrypt(cipher, key); // 对称
}

int main() {
  string text = "HelloAES";
  string key = "key123";

  string enc = aes_encrypt(text, key);
  string dec = aes_decrypt(enc, key);

  cout << "Plain: " << text << endl;
  cout << "Encrypted: " << enc << endl;
  cout << "Decrypted: " << dec << endl;
}